File: btpy/src/btpy/btpy_loops/mod/has_all/has_all.py



def has_all(ARRAY:list|dict|str, function):
    """
    Función que itera sobre los elementos del
    array enviado buscando si todos los 
    elementos del array retornan true a la 
    función enviada.
    """
    data = None
    if(type(ARRAY) == dict):
        data = ARRAY.values()
    else:
        data = ARRAY
    for e in data:
        if(not function(e)):
            return False
    return True

File: btpy/on_work/structures/has_all.js


import { Btjs } from "../../Btjs.js"

/*
Función que itera sobre los elementos del
array enviado buscando si todos los 
elementos del array retornan true a la 
función enviada.
*/
export function has_all(ARRAY, 
        CHECK_FUNCTION) {
    let data = null
    if(Btjs.is_object_js(ARRAY)) {
        data = ARRAY.values()
    }
    else {
        data = ARRAY
    }
    for(let i in data) {
        if( ! CHECK_FUNCTION(data[i])) {
            return false
        }
    }
    return true
}
    